packages/zowe-explorer/src/dataset/init.ts

```diff
@@ -13,18 +13,29 @@ import * as vscode from "vscode";
 import * as globals from "../globals";
 import * as dsActions from "./actions";
 import * as refreshActions from "../shared/refresh";
-import { IZoweDatasetTreeNode, IZoweTreeNode, ZosEncoding } from "@zowe/zowe-explorer-api";
+import { IZoweDatasetTreeNode, IZoweTreeNode, ZosEncoding, ZoweScheme } from "@zowe/zowe-explorer-api";
 import { Profiles } from "../Profiles";
 import { DatasetTree, createDatasetTree } from "./DatasetTree";
 import { ZoweDatasetNode } from "./ZoweDatasetNode";
 import * as contextuals from "../shared/context";
-import { getSelectedNodeList } from "../shared/utils";
+import { getLanguageId, getSelectedNodeList } from "../shared/utils";
 import { initSubscribers } from "../shared/init";
 import { ZoweLogger } from "../utils/ZoweLogger";
 import { TreeViewUtils } from "../utils/TreeViewUtils";
+import { DatasetFSProvider } from "./DatasetFSProvider";
+import { posix as posixPath } from "path";
 
 export async function initDatasetProvider(context: vscode.ExtensionContext): Promise<DatasetTree> {
     ZoweLogger.trace("dataset.init.initDatasetProvider called.");
+
+    context.subscriptions.push(vscode.workspace.registerFileSystemProvider(ZoweScheme.DS, DatasetFSProvider.instance, { isCaseSensitive: true }));
+    context.subscriptions.push(
+        vscode.commands.registerCommand(
+            "zowe.onDatasetChanged",
```

Comment:
Curious if rather than having a separate command for each tree, we could use a single command `zowe.onResourceChanged` that takes the resource type (dataset, ussFile, job) as an argument?

Or perhaps instead of using commands, we can expose the `onDidChangeFile` events on the `ZoweExplorerExtenderApi` class that extenders are able to import?

Sorry if I'm being picky - this seems like a non-standard usage of VS Code commands to register events so wondering if we can avoid it 😋

Reply:
I've implemented a combination of your feedback by adding the `onResourceChanged` function in `ZoweExplorerApiRegister` that takes in a `ZoweScheme` as the parameter - it will return the corresponding event.

Thanks for the suggestions!